airbyte-workers/src/main/java/io/airbyte/workers/DefaultCheckConnectionWorker.java

```diff
@@ -77,7 +77,7 @@ public StandardCheckConnectionOutput run(final StandardCheckConnectionInput inpu
         LOGGER.debug("Check connection job received output: {}", output);
         return output;
       } else {
-        throw new WorkerException("Error while getting checking connection.");
+        throw new WorkerException(String.format("Error checking connection, status: %s, exit code: %d", status, exitCode));
```

Comment:
@jrhizor, a check operation failed in the acceptance test. I added more information to the exception message and saw this:

```
Error checking connection, status: Optional[io.airbyte.protocol.models.AirbyteConnectionStatus@605d010e[status=SUCCEEDED,message=<null>,additionalProperties=***]], exit code: 143
```

Do you know what might be the root cause behind this exit code? Could it be related to the recent change in `WorkerUtils#gentleClose`?

Reply:
I don't think the gentle close changes look related. Are you sure this docker container isn't actually outputting an error code?

Reply:
Here is what happens:
- The database connection somehow requires an explicit `close` method call, otherwise, the connector will not shutdown.
- The worker will wait for 1 minute for the connector to shutdown, otherwise it will close it by force (code [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-workers/src/main/java/io/airbyte/workers/WorkerUtils.java#L53)) by sending a `SIGTERM`.
- The `SIGTERM` results in a 143 exit code.

So the root cause is that we must always close the database at the end, which was not the case in the default check command implementation in `CopyDestination`. The issue has been resolved.